presto-main/src/main/java/com/facebook/presto/operator/HashSemiJoinOperator.java

```diff
@@ -174,7 +174,12 @@ public void addInput(Page page)
         // update hashing strategy to use probe cursor
         for (int position = 0; position < page.getPositionCount(); position++) {
             if (probeJoinPage.getBlock(0).isNull(position)) {
-                blockBuilder.appendNull();
+                if (channelSet.size() > 0) {
```

Comment:
will not work in distibution. As channelSet contains only subset of data.

As we discussed. I am not sure if it is worth an effort. If it is not trivial.